### PR TITLE
Feature/lior/73 layout redirection by language

### PR DESCRIPTION
**Pull request by LiorSil**
- Fixed redirection of UI layout depends on the selected language 
- Implemented tFunction from useTranslation (i18n) for Settings Modal
- Add translations for the Settings Modal in the translation.json :

1. langauge:  Language / "שפה"
2. theme: Theme / "ערכת נושא"

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -2,10 +2,12 @@ import { useAuth0 } from "@auth0/auth0-react";
 import { Dashboard, NavBar } from "./components";
 import { QueryWrapper, useTheme } from "./components/utilities";
 import "./App.css";
+import i18n from "./services/translation";
```
Comment (ofir-zeitoun) [inline src/App.tsx:5]: reorder

File: src/App.tsx [hunk 2]
```diff
@@ -2,10 +2,12 @@ import { useAuth0 } from "@auth0/auth0-react";
...
 
 function App() {
   const { isAuthenticated } = useAuth0();
   useTheme();
+  console.log("i18n-direction", i18n.dir());
```
Comment (ofir-zeitoun) [inline src/App.tsx:10]: remove

File: src/components/translation/use-translation.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+      LanguageOption[0].label,
+    [i18nLanguage]
+  );
+
+  
+
+  const changeLanguage = useCallback(
+    (language: string) => {
+      i18n.changeLanguage(language);
+      document.body.dir = i18n.dir(language);
```
Comment (ofir-zeitoun) [inline src/components/translation/use-translation.ts:22]: use useEffect for updating dom
use state for LanguageDir = i18n.dir(language);

File: src/components/translation/use-translation.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+  const i18nLanguage = i18n.language.split("-")[0];
+
+  const currentLanguage = useMemo(
+    () =>
+      LanguageOption.find((lang) => lang.key === i18nLanguage)?.label ||
+      LanguageOption[0].label,
+    [i18nLanguage]
+  );
+
```
Comment (ofir-zeitoun) [inline src/components/translation/use-translation.ts:16]: remove spaces

File: src/components/modals/settings-modal/settings-modal.tsx [hunk 5]
```diff
@@ -2,13 +2,15 @@ import { Modal } from "antd";
 import { ThemeSwitcher } from "../../theme-switcher";
 import { TranslationSwitcher } from "../../translation";
 import "./settings-modal.css";
+import { useTranslation } from "react-i18next";
```
Comment (ofir-zeitoun) [inline src/components/modals/settings-modal/settings-modal.tsx:5]: use 'our' useTranslation

File: src/App.tsx (hunk 1 above)
Comment (LiorSil) [reply src/App.tsx:5]: done

File: src/components/translation/use-translation.ts (hunk 3 above)
Comment (LiorSil) [reply src/components/translation/use-translation.ts:22]: removed unnecessary consoloe.log()

File: src/components/translation/use-translation.ts (hunk 4 above)
Comment (LiorSil) [reply src/components/translation/use-translation.ts:16]: done

File: src/components/modals/settings-modal/settings-modal.tsx (hunk 5 above)
Comment (LiorSil) [reply src/components/modals/settings-modal/settings-modal.tsx:5]: done

**Review by ofir-zeitoun — APPROVED** [review]

**Pull request merged**